src/vs/workbench/contrib/surveys/browser/ces.contribution.ts

```diff
@@ -0,0 +1,137 @@
+import { language } from 'vs/base/common/platform';
+import { IWorkbenchContributionsRegistry, IWorkbenchContribution, Extensions as WorkbenchExtensions } from 'vs/workbench/common/contributions';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+import { IStorageService, StorageScope, StorageTarget } from 'vs/platform/storage/common/storage';
+import { IProductService } from 'vs/platform/product/common/productService';
+import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle';
+import { Severity, INotificationService } from 'vs/platform/notification/common/notification';
+import { IOpenerService } from 'vs/platform/opener/common/opener';
+import { ITASExperimentService } from 'vs/workbench/services/experiment/common/experimentService';
+import { URI } from 'vs/base/common/uri';
+import { platform } from 'vs/base/common/process';
+import { RunOnceScheduler } from 'vs/base/common/async';
+import { Disposable } from 'vs/base/common/lifecycle';
+
+const WAIT_TIME_TO_SHOW_SURVEY = 1000 * 60 * 60; // 1 hours
+const MAX_INSTALL_AGE = 1000 * 60 * 60 * 24; // 24 hours
+const REMIND_LATER_DELAY = 1000 * 60 * 60 * 4; // 4 hours
+const SKIP_SURVEY_KEY = 'ces/skipSurvey';
+const REMIND_LATER_DATE_KEY = 'ces/remindLaterDate';
+
+class CESContribution extends Disposable implements IWorkbenchContribution {
+	private promptScheduler!: RunOnceScheduler;
+
+	constructor(
+		@IStorageService private readonly storageService: IStorageService,
+		@INotificationService private readonly notificationService: INotificationService,
+		@ITelemetryService private readonly telemetryService: ITelemetryService,
+		@IOpenerService private readonly openerService: IOpenerService,
+		@IProductService private readonly productService: IProductService,
+		@optional(ITASExperimentService) private readonly tasExperimentService: ITASExperimentService,
+	) {
+		super();
+
+		if (!productService.cesSurveyUrl) {
+			return;
+		}
+
+		const skipSurvey = storageService.get(SKIP_SURVEY_KEY, StorageScope.GLOBAL, '');
+		if (skipSurvey) {
+			return;
+		}
+		this.promptScheduler = this._register(new RunOnceScheduler(async () => {
+			const sendTelemetry = (userReaction: 'accept' | 'remindLater' | 'neverShowAgain') => {
+				/* __GDPR__
+				"cesSurvey:popup" : {
+					"userReaction" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+				}
+				*/
+				this.telemetryService.publicLog('cesSurvey:popup', { userReaction });
+			};
+
+			const message = await this.tasExperimentService?.getTreatment<string>('CESSurveyMessage') ?? nls.localize('cesSurveyQuestion', 'Got a moment to help the VS Code team? Please tell us about your experience with VS Code so far.');
+			const button = await this.tasExperimentService?.getTreatment<string>('CESSurveyButton') ?? nls.localize('giveFeedback', "Give Feedback");
+
+			this.notificationService.prompt(
```

Comment:
The result of `prompt()` should be stored in a variable, such that we can `await` for the prompt to be closed and make sure the `RunOnceScheduler` does its job of just running this one prompt at a time.